src/context/ArchiveContext/ArchiveContext.jsx

```diff
@@ -0,0 +1,71 @@
+import { useContext, createContext, useState, useEffect } from 'react';
+import { useAxios } from '../../CustomHooks';
+
+const ArchiveContext = createContext();
+
+const ArchiveProvider = ({ children }) => {
+	const [archiveNotes, setArchiveNotes] = useState([]);
+	const { response: archiveResponse, operation: fetchArchive } = useAxios();
+
+	useEffect(() => {
+		if (archiveResponse != null) {
+			if (archiveResponse.archives) {
+				setArchiveNotes(archiveResponse.archives);
+			}
+		} else {
+			getArchive();
+		}
+	}, [archiveResponse]);
+	const getArchive = () => {
+		fetchArchive({
+			method: 'GET',
+			url: '/api/archives',
+			headers: {
+				authorization:
+					'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI0OGJmMzFiYi1iYWQyLTQ3N2EtYmRhNy1kZDc4NDIxNDZhODgiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.kH6xXhMMvQ0sgsCz7e8vrSh1pVddplgC3IMzUSlTFsQ',
+			},
+		});
+	};
+	const notesToArchive = (noteId, notes) => {
+		fetchArchive({
+			method: 'POST',
+			url: `/api/notes/archives/${noteId}`,
+			data: {
+				notes,
+			},
+			headers: {
+				authorization: localStorage.getItem('token'),
```

Comment:
Maybe from next time you can also pass the token to this function. Reason being you should not have to access the local storage repeatedly for getting the encoded token. Especially when you are already storing that data. In short, avoid accessing local storage unless you are initializing somewhere or setting the data.

Reply:
will refractor this in the upcoming pr.